build.rs

```diff
@@ -54,24 +200,48 @@ to MKL modules that you would like to use, or enable the `all` feature if you wo
 like to generate symbols for all modules.");
     }
 
-    let name = build_config_name();
-    let library = pkg_config::probe_library(&name).unwrap();
-
     // Use information obtained from pkg-config to setup args for clang used by bindgen.
     // Otherwise we don't get e.g. the correct MKL preprocessor definitions).
     let clang_args = {
-        let mut args = Vec::new();
-        for (key, val) in library.defines {
-            if let Some(value) = val {
-                args.push(format!("-D{}={}", key, value));
-            } else {
-                args.push(format!("-D{}", key));
+        let name = build_config_name();
+
+        match pkg_config::probe_library(&name) {
+            // First try using pkg-config
+            Ok(library) => {
+                let mut args = Vec::new();
+                for (key, val) in library.defines {
+                    if let Some(value) = val {
+                        args.push(format!("-D{}={}", key, value));
+                    } else {
+                        args.push(format!("-D{}", key));
+                    }
+                }
+                for path in library.include_paths {
+                    args.push(format!("-I{}", path.display()));
+                }
+                args
+            }
+            // Otherwise use hardcoded paths relative to MKLROOT
+            Err(_) => {
+                let mklroot = match env::var("MKLROOT") {
+                    Ok(mklroot) => mklroot,
+                    Err(_) => panic!("Environment variable 'MKLROOT' is not defined."),
```

Comment:
Would be nice if we would also explain that `pkg-config` also failed.. We should also discuss if we should *prefer* `MKLROOT` over `pkg-config`, as this would perhaps make it easier to override MKL for a certain build?

Reply:
I revised the error message when both methods fail.

In my opinion the current order of checks is ok, as the pkg-config search path can be extended using an environment variable. Using this approach however it is admittedly not possible to force the build script into the non-pkg-config branch if pkg-config can find something (I think).